File: VB/GridEntityFramework/App_Code/Product.vb
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.ComponentModel.DataAnnotations

Namespace GridEntityFramework

    Public Partial Class Product

        Public Property ProductID As Integer

        <Required>
        <StringLength(40)>
        Public Property ProductName As String

        Public Property SupplierID As Integer?

        Public Property CategoryID As Integer?

        <StringLength(20)>
        Public Property QuantityPerUnit As String

        <Column(TypeName:="money")>
        Public Property UnitPrice As Decimal?

        Public Property UnitsInStock As Short?

        Public Property UnitsOnOrder As Short?

        Public Property ReorderLevel As Short?

        Public Property Discontinued As Boolean

        Public Overridable Property Category As Category
    End Class
End Namespace

File: CS/GridEntityFramework/App_Code/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Data.Entity.Spatial;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace GridEntityFramework
{
    public partial class Product
    {
        public int ProductID { get; set; }
        [Required]
        [StringLength(40)]
        public string ProductName { get; set; }
        public int? SupplierID { get; set; }
        public int? CategoryID { get; set; }
        [StringLength(20)]
        public string QuantityPerUnit { get; set; }
        [Column(TypeName = "money")]
        public decimal? UnitPrice { get; set; }
        public short? UnitsInStock { get; set; }
        public short? UnitsOnOrder { get; set; }
        public short? ReorderLevel { get; set; }
        public bool Discontinued { get; set; }
        public virtual Category Category { get; set; }
    }
}